engine/src/main/java/io/zeebe/engine/processing/bpmn/container/CallActivityProcessor.java

```diff
@@ -65,63 +65,38 @@ public void onActivating(final ExecutableCallActivity element, final BpmnElement
         .flatMap(this::checkProcessHasNoneStartEvent)
         .ifRightOrLeft(
             process -> {
-              stateTransitionBehavior.transitionToActivated(context);
+              final var activated = stateTransitionBehavior.transitionToActivated(context);
 
               final var childProcessInstanceKey =
                   stateTransitionBehavior.createChildProcessInstance(process, context);
 
-              final var callActivityInstance = stateBehavior.getElementInstance(context);
-              callActivityInstance.setCalledChildInstanceKey(childProcessInstanceKey);
```

Comment:
Do we not need to set the called child instance key anymore? I see we do it in the applier, but wouldn't that only work if process is migrated? AFAIK right now it will just write a follow up event but won't apply the state change?

Reply:
While process is not yet migrated, this is achieved in `BpmnStreamProcessor`. It's done when (re)processing the Process:Activating for a child process. By always doing it there, processing and reprocessing result in the same state.